x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java

```diff
@@ -162,6 +161,51 @@ public void testCloseAndReopen() throws Exception {
         });
     }
 
+    public void testStats() throws Exception {
+        runSearchableSnapshotsTest((restoredIndexName, numDocs) -> {
+            final Map<String, Object> stats = searchableSnapshotStats(restoredIndexName);
+            assertThat("Expected searchable snapshots stats for [" + restoredIndexName + ']', stats.size(), greaterThan(0));
+
+            final int nbShards = Integer.valueOf(extractValue(indexSettings(restoredIndexName), IndexMetaData.SETTING_NUMBER_OF_SHARDS));
+            assertThat("Expected searchable snapshots stats for " + nbShards + " shards but got " + stats, stats.size(), equalTo(nbShards));
+        });
+    }
+
+    public void testClearCache() throws Exception {
+        @SuppressWarnings("unchecked")
+        final Function<Map<?, ?>, Long> sumCachedBytesWritten = stats -> stats.values().stream()
+            .filter(o -> o instanceof List)
+            .flatMap(o -> ((List) o).stream())
+            .filter(o -> o instanceof Map)
+            .map(o -> ((Map<?,?>)o).get("files"))
+            .filter(o -> o instanceof List)
+            .flatMap(o -> ((List) o).stream())
+            .filter(o -> o instanceof Map)
+            .map(o -> ((Map<?,?>)o).get("cached_bytes_written"))
+            .filter(o -> o instanceof Map)
+            .map(o -> ((Map<?,?>)o).get("sum"))
+            .mapToLong(o -> ((Number) o).longValue())
+            .sum();
```

Comment:
I wonder whether it'd be better to do this whole test in the YAML suite. I think that's possible. WDYT?

Reply:
I'm not sure I can do it in YAML: this test computes the total bytes written in cache (the stats API does not compute total among all shards and files) after a given query, then clear the cache and run the query again to verify that more data has been written in cache gain.

I pushed 587819a2c8cbdd0048c4c4c24a762367ea0e5969 to run the same query before/after the cache clearance.

Reply:
Oh yes so it does. Let's not add  `.sum()` to the YAML test runner :)